CRM: Fix export crash when custom field name clashes with linked objects

gogdzl opened the pull request:
Fixes https://github.com/Automattic/zero-bs-crm/issues/3477

## Proposed changes:

* This PR fixes a fatal error that occurs when exporting contacts with a custom field that had the same name as a linked field (e.g., 'company').
* The way this works is by adding a suffix to a key when there is a name clash, then before the export process uses custom field keys, it checks for clashes. The former is a manual check that has to be added in every place we may have a name clash.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
https://github.com/Automattic/zero-bs-crm/issues/3477

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

Steps to replicate this error:

1. Add a custom text field named 'Company' to contacts
2. Add some information to this new custom field for at least one contact
3. Try to export the contact list ( wp-admin/admin.php?page=zbs-export-tools) - Make sure the custom field is selected

The export process will crash.
With this PR the crash no longer happens.

gogdzl commented on projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Contacts.php:
```diff
@@ -2114,25 +2134,25 @@ public function getContacts($args=array()){
-									)
-								);
+													)
+												);
```
Added this whole chunk of tabs to make the linter happy.

cleacos commented on projects/plugins/crm/includes/ZeroBSCRM.DAL3.ObjectLayer.php:
```diff
@@ -44,6 +44,18 @@ class zbsDAL_ObjectLayer {
+	/** Suffix used to fix name clashes
+	 * See: https://github.com/Automattic/zero-bs-crm/issues/3477
+	 */
+	protected const NAME_CLASH_FIX_SUFFIX = '_zbs-avoid-name-clash-temporary-fix';
```
The suffix seems large. Could we use something shorter? Maybe a numeric suffix could work. I'm thinking of the usage of the field name, like in templates.

gogdzl replied:
Made it a little shorter in e5ce479

coder-karen commented on projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Contacts.php:
```diff
@@ -2072,6 +2072,26 @@ public function getContacts($args=array()){
+										'dnd',
+										'company',
+										'lastlog',
+										'owner',
```
I've been looking at the PHP test failures related to ` Undefined array key "owner" in /tmp/wordpress-latest/src/wp-content/plugins/crm/includes/ZeroBSCRM.Edit.php on line 185`, and I was able to replicate the error in my logs locally by creating a new invoice and assigning it to a contact .

On logging `$data['invoiceObj']['contact']`, there was no `owner` key but there was an `owner_zbs-avoid-name-clash-temporary-fix` key. 

If I remove `owner` from the array passed in to `add_name_clash_suffix_if_needed`, I get no more warnings.
Should the `owner` have a prefix attached to it?

gogdzl replied:
Thanks for catching this! Probably it needs some more conditions to prevent adding suffix to Jetpack CRM fields, we want only to add the suffix to custom ones.

gogdzl replied:
Fixed in c8e6376. It is clumsy but it will always work because we are calling `fix_name_clash_if_needed` only in our export method. It would probably fail if we have two custom fields with the same key, what should not happen.

coder-karen commented on projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Contacts.php:
```diff
@@ -2072,6 +2072,26 @@ public function getContacts($args=array()){
+										'company',
+										'lastlog',
+										'owner',
+										'company',
```
Should company be listed here twice?

gogdzl replied:
No!

gogdzl replied:
Fixed in 775eb5d

coder-karen reviewed (changes requested):
Thanks for tackling this!  I left a few questions after looking specifically at the test failures.

anomiex committed:
docker: Re-add xz-utils (#39553)

Previously this was installed as a dependency of mailcap, which was a
dependency of mime-support, which was a dependency of various things.
But apparently things depending on mime-support are transitioning to
media-types which doesn't have that dependency.

We need xz-utils for extracting `build.tar.xz` in CI, so make sure it's
installed.

ilyasfoo committed:
Bump wc-calypso-bridge to 2.7.1 and add changelog (#39557)

CodeyGuyDylan committed:
Add/icons for all product cards on dashboard (#39513)

* Ensure all product cards on Jetpack Dashboard have icons when inactive

* changelog

* Add icon to boost install section

* Fix AI missing icon when active

* Move CRM back to where it was

CodeyGuyDylan committed:
Remove/connection nudges where not needed (#39533)

* Remove connection nudges for cards that don't need it on dashboard

* Ask for user connection on Blaze when not present and remove nudge on Akismet

* changelog

* Get rid of connection nudge for Search modules

* Fix blaze tests

matticbot committed:
phan: Update wpcom stubs (#39565)

Co-authored-by: Phabricator Bot <phabricator@automattic.com>

mreishus committed:
wpcom_add_shopping_cart: Use new is_cart_empty function (#39563)

kraftbj committed:
Blocks: convert to module (#39449)

* Blocks: convert to a module

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

nateweller committed:
Jetpack: Remove WAF deprecation notice (#39568)

lhkowalski committed:
Title Optimization: improve button behavior on error screen (#39554)

* Properly close the modal when Cancel is clicked, cleaning up the state

* Skip new generation on try again button when there are options already

* Changelog

sergeymitr committed:
Jetpack: remove excessive hooks from social logos and genericons (#39564)

`jetpack_register_social_logos()` and `jetpack_register_genericons` is called directly, and the files are [loaded](https://github.com/Automattic/jetpack/blob/8d8e0e6ef75b32c52029a57740cb6e13c6312296/projects/plugins/jetpack/class.jetpack.php#L735) on `wp_loaded` after the `init` hook has already run.

d-alleyne committed:
Remove/unnecessary textdomain filter (#39552)

* i18n: Remove unnecessary filter to loading translations

This filter has been moved to D162485-code

* changelog

* Add changelog files

CGastrell committed:
Jetpack AI Logo generator toolbar button label (#39550)

* add prop to toggle button text

* show button text Generate if the site has no logo set

* add changelog

* rename prop so it's not misleading

CodeyGuyDylan committed:
Change/at a glance unify connection ctas (#39535)

* Update CTA for connection related cards on At A Glance

* changelog

* Fix JS tests

* Fix e2e test

* Fix alignment issue on smaller screens

* Fix e2e test

* Fix cta spacing issue

* Add border to connection cards

* Fix e2e test

* Make border transparent

* Align content with title on connection cards

sergeymitr committed:
Jetpack: remove unused infinite scroll code (#39570)

* Remove unused and non-working code.
* Deprecate functions that are no longer needed to be removed later.

dkmyta committed:
Protect: Fix blockedRequests destructuring fallback (#39569)

dkmyta committed:
Protect: Fix overwritten FirewallSubheading updates (#39571)

* Reapply overwritten changes

* changelog

* Fix import

* Move stories to broken directory

IanRamosC committed:
My Jetpack: update notice style on mobile (#39537)

* My Jetpack: update notice mobile style

* changelog

* Update notice style to the same when there's title

elliottprogrammer committed:
My Jetpack Welcome Flow: Show default recommendations upfront first, then offer optional survey (#39485)

fgiannar committed:
Jetpack plugin - Story Block: Fix 'Division by zero' Fatal (#39577)

* Jetpack plugin - Story Block: Fix 'Division by zero' Fatal

fgiannar committed:
Masterbar: Remove User Info side-panel (#39546)

* Masterbar: Remove User Info side-panel

fgiannar committed:
Jetpack Sync: Fix a bug in syncing HPOS 'woocommerce_delete_order' actions (#39562)

* Jetpack Sync: Fix a bug in syncing HPOS 'woocommerce_delete_order' actions

elliottprogrammer committed:
'connect_url_redirect' hook: Redirect to the 'redirect_after_auth' url if already connected (for 'connect_after_checkout' flow) (#39573)

robertsreberski committed:
Fix parameters to allow for connectAfterCheckout flow from recommendation card (#39578)

* Fix parameters to allow for connectAfterCheckout flow from recommendation card

* changelog

* Fix fetching the default value in the hook itself

* changelog

arcangelini committed:
Help Center: use Zendesk staging when proxied (#39566)

IanRamosC committed:
My Jetpack: fix spacing between price and discount tag (#39576)

* My Jetpack: fix discount tag spacing

* changelog

Aurorum committed:
Follow Blog Widget: Strip Slashes on Subscribe Text (#39433)

Removes slashes from the Subscribe text output on the Follow Blog
widget. It looks like the Subscribe block already does this, which is
why the issue doesn't happen there. It's just been annoying me a bit
lately that the Follow Blog widget doesn't do the same too.

zinigor committed:
Fix a case where wp_rand is undefined. (#39581)

* Added a check for wp_rand in case pluggables are not loaded yet.

* Changelog.

CodeyGuyDylan committed:
Revert "Change/at a glance unify connection ctas (#39535)" (#39584)

This reverts commit 4f47d298ffd8726cc5168a27d5540e0242388755.

anomiex committed:
wpcomsh: Move `load_muplugin_textdomain` call to `after_setup_theme` hook (#39586)

manzoorwanijk committed:
Social: Add Bluesky connection UI (#39561)

* Add Bluesky to the supported services

* Update connect URL logic for Bluesky

* Create a dedicated component for custom inputs

* Update ConnectForm to use CustomInputs

* Update ServiceItem component to handle Bluesky custom inputs

* Do the make up

* Add changelog

* Fix messed up translations

* Clean up

zinigor committed:
Backport jetpack 13.9-beta Changes (#39587)

* Changelog and readme.txt edits.

* Amend readme.txt

gmjuhasz committed:
Changelog and readme.txt edits. (#39598)

manzoorwanijk committed:
Social: Fix Bluesky not showing up on page load (#39597)

simison committed:
Email preview: improve error state visually (#39476)

ecairol committed:
Jetpack: Check $submenu is not NULL before adding Scan option (#39424)

* Check $submenu is not NULL before accessing it

* Add changelog

---------

Co-authored-by: mmtr <1233880+mmtr@users.noreply.github.com>

CodeyGuyDylan committed:
Update/at a glance unify connection ctas (#39585)

* Unify connection notices on At A Glance

* changelog

* Fix nudges on settings page

* Fix tests

sbarbosa committed:
Changes the error Blaze shows during sync to be a warning (#39515)

* Changes the error Blaze shows during sync to be a warning

CodeyGuyDylan committed:
Update/jetpack settings connection nudges (#39591)

* Unify connection nudges on settings

* changelog

matticbot committed:
Update dependency core-js to v3.38.1 (#39592)

* Update dependency core-js to v3.38.1
* Remove unnecessary deps

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

zinigor committed:
Changelog. (#39606)

valterlorran committed:
Post-migration Experience: Adds the placeholder task list (#39593)

* Add the placeholder Migration task list

* changelog

* Update the tasklist name to match calypso's

---------

Co-authored-by: Valter Lorran <valter.lorran@automattic.com>

zinigor committed:
Release WAF 0.20.1. (#39607)

matticbot committed:
Update Size-limit to v11.1.6 (#39595)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency uplot to v1.6.31 [SECURITY] (#39610)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

dkmyta committed:
Protect: Fix last_checked conversions (#39549)

* Convert last_checked value so it in local time before the WP install timezone conversions apply

* changelog

* Update var naming

dkmyta committed:
Protect: Improve optimistically scanning (#39538)

* Improve optimistically scanning

* Update formatting

* changelog

* Always return API data when scan status is non-idle

* Fix lastChecked timezone conversion

* Clean up useScanStatusQuery queryFn

* Update projects/plugins/protect/src/js/data/scan/use-scan-status-query.ts

Co-authored-by: Nate Weller <nate.weller@automattic.com>

* Fix lint error

* Fix lint error

* Update projects/plugins/protect/src/js/data/scan/use-scan-status-query.ts

Co-authored-by: Nate Weller <nate.weller@automattic.com>

* Update projects/plugins/protect/src/js/data/scan/use-start-scan-mutation.ts

Co-authored-by: Nate Weller <nate.weller@automattic.com>

---------

Co-authored-by: Nate Weller <nate.weller@automattic.com>

nateweller committed:
Protect: Minor clean up of scan page index component (#39574)

grzegorz-cp committed:
Settings: Newsletter: Adding connection banner to each section (#39539)

* Settings: Adding connection banner for all Newsletter options

* changelog

* Settings: Updating icon for the Newsletter connection banner

* Settings: Updating icon for the Newsletter connection banner

* Settings: Renaming const and its value

coder-karen committed:
Admin Menu: Adding tests to test Jetpack admin menu 'current' menu item class and menu order (#39374)

candy02058912 committed:
Update Jetpack Scan link (#39619)

* Update Jetpack Scan link in menu

* changelog

* Update Jetpack scan link for Default view

* changelog

* Fix test

yuliyan committed:
wpcomsh: Update translations (#39621)

* Update language files

* changelog

yuliyan committed:
wpcomsh: Update translations (#39623)

* Update language files

* changelog

zinigor committed:
Backport Jetpack 13.9 changes to readme and CHANGELOG.md. (#39622)

* Changelog edits.

* Changelog squash and readme.txt.

* Jetpack 13.9.

* Reset to-test.md.

gmjuhasz committed:
Changelog and readme.txt edits. (#39624)

sergeymitr committed:
Connection: initialize assets in wp-admin only (#39604)

Only initialize Connection assets in the admin area.

matticbot committed:
Update wordpress monorepo (#39594)

* Update wordpress monorepo
* Update test for Modal close animation
* Remove obsolete `@types/wordpress__media-utils`, as `@wordpress/media-utils` contains types now. Fix one type.

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

kraftbj committed:
Performance: remove imploding CSS code (#39518)

dkmyta committed:
Protect: Update use fixers query error prop handling (#39498)

* Update FixerStatus type to account for potential error props

* changelog

* Handle error responses

* Only handle top level errors in useFixersQuery, threat level errors are handled in the completion block

* Add clarifying comment

* Improve fixer types

* Fix prop handling

* Update useFixers hook for compatibility with type updates

* Remove optional chaining where applicable, update logic to rely on TS types

* Update types

* Fix type error

---------

Co-authored-by: Nate Weller <hello@nateweller.com>

mreishus committed:
wpcomsh: Update wc-calypso-bridge to 2.8.0 (#39613)

oskosk committed:
Sitemap: Make sitemap storing more efficient by not querying the full post content before saving (#39572)

* Make sitemap storing more efficient but not querying the full post content when storing the sitemap

This update tries to remediate a problem where the persistent object cache gets huge
with data that we will not use in the end.
We just need to get the id of the current sitemap post and if it exists, we update it.

Before, we were using get_post and this polluted the cache running into OOM situations

darssen committed:
Shortcodes: Fix warning when attachment src is false in shortcodes module (#39601)

* Check if the attachment is false, to avoid warnings

haqadn committed:
Boost: Exclude shortcode produced scripts from deferred js (#39616)

* Exclude shortcode provided scripts from deferred js

* changelog

darssen committed:
Roles endpoint: Use instance of for checking wp roles (#39634)

* Improve roles check

* changelog

* Added defensive check for arrays

* Updated phan baseline

* Jetpack plugin: Update phan baselines

---------

Co-authored-by: Foteini Giannaropoulou <giannaropoulou.foteini@gmail.com>

matticbot committed:
phan: Update custom stubs (#39632)

Co-authored-by: matticbot <matticbot@users.noreply.github.com>

jeherve committed:
Issue templates: add new issue type option (#39637)

This was added as an option as part of this set of changes to GitHub issues:
https://github.blog/changelog/2024-10-01-evolving-github-issues-public-preview/

matticbot committed:
Update dependency photon to v4.1.1 (#39639)

myhro committed:
blogging-prompt: preserve language on answers link (#36730)

fushar committed:
Admin bar: replace 'Edit Profile' and 'My Account' with 'My Profile' (#39600)

fushar committed:
profile.php: Make wpcom links more concise (#39636)

manzoorwanijk committed:
Social: Fix share status tooltip text overflow (#39599)

okmttdhr committed:
WPCOMSH: Remove redirection from /wp-admin/plugin.php to /plugins/manage/<site> (#39615)

* remove redirection

* changelog

haqadn committed:
Boost: Fix photon URL check (#39635)

* Extract is_photon_url to a method

* Improve photon URL check in image guide proxy

* changelog

* Rename the method for consistency

* Add boost changelog

* Add a test

* Update projects/packages/image-cdn/src/class-image-cdn-core.php

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

* Update projects/packages/image-cdn/changelog/add-photon-url-check-method

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

---------

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

matticbot committed:
Update dependency @sentry/browser to v8 [SECURITY] (#39640)


Co-authored-by: Renovate Bot <bot@renovateapp.com>

jeherve committed:
e2e tests: update encryption key (#39645)

CodeyGuyDylan committed:
Update VideoPress block connection banner, align with other blocks (#39628)

* Make VideoPress connection banner consistent with Nudge component

* changelog

* Polish the VideoPress connection banner

* Move constants in block

* Remove unintentional change

nateweller committed:
WAF: Allow rules to specify body parser type (#39516)

* WAF: Add support for rules file to manually specify body parsing method

* Add public $version property to Waf_Runtime

kraftbj committed:
Blocks: provide nice error message for workaround in dev env (#39602)

* Provide nice error message for workaround

* typos will be the dealth of me

grzegorz-cp committed:
Settings - Newsletter, Sharing: Disabling options for sites missing user connection (#39520)

* Newsletters: Disabling controls that need user connection

* changelog

* Settings: Fixing disabled label when not saving

* Settings: Improving isLinked check

* Settings: Removing disabled CSS from active controls

* Settings: Disabling sharing without a connected user

* Settings: Disabling newsletter settings for sites without a single connected user

* Settings: Disabling also whole groups

* Settings: Updating the description of the task

* Settings: Updating the description of the task

dilirity committed:
Boost: Use "critical css gen" package from the monorepo (#39509)

* Update Boost to use "critical css gen" package from the monorepo

* add changelog

* Update Boost to use new browser and node from "critical css gen" package

* add changelog

* Update package entry points

* Cleanup package entry points

* Use dynamic import for global type definitions

* Add browser entry point back to package

* Use dynamic import for css generator

* Don't use destructuring

* Update default entry point to use browser version

* Update Boost to use updated package entry point

* Simplify package build

* Remove unnecessary dependency enqueue

* Attempt to fix tests

* Rename dynamically imported file

* Merge dynamic import files

* Add build step for tests

* Cleanup build for tests

* Fix tests breaking due to missing package build

* Update package entry points

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

* Add helper for critical css generator import

* update changelogs

---------

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

valterlorran committed:
Post-migration Experience: Add the "Review the site's content" task (#39641)

* Add the Review site task

* changelog

* Remove calypso path

---------

Co-authored-by: Valter Lorran <valter.lorran@automattic.com>

dilirity committed:
Backport changes in JS Packages (#39664)

* Update versions/changelog

* Update lock file

lhkowalski committed:
VideoPress: use "Processing" thumbnail placeholder while a new thumbnail is being saved (#39631)

* Export the thumb update status on the edit details hook

* Use poster updating status to set video thumb as processing

* Changelog

---------

Co-authored-by: Douglas <douglas.henri@automattic.com>

sergeymitr committed:
Connection: optimize admin_notices owner delete warning (#39650)

Optimize the `admin_notices` hook conditions to get rid of an excessive callback.

anomiex committed:
Fix new PHPCompatibility-dev sniff (#39665)

PHPCompatibility, in their dev branch, just added a sniff for the
`$escape` parameter to various CSV-related functions, since PHP has
decided they might change the default in 9.0.

matticbot committed:
Update dependency cookie to v0.7.0 [SECURITY] (#39649)

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency mediawiki/mediawiki-codesniffer to v44 (#39648)

* Update dependency mediawiki/mediawiki-codesniffer to v44
* Update monorepo root composer.lock. New versions of various other phpcs packages.
* MediaWiki.Usage.DoubleNotOperator has been removed upstream.
* Disable Generic.CodeAnalysis.RequireExplicitBooleanOperatorPrecedence.MissingParentheses pending discussion
* Fix new Squiz.Commenting.FunctionComment.MissingParamType sniffs

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

CGastrell committed:
Jetpack AI: change logo generator styles source (#39589)

* remove style definitions from frontend, add type for style object

* update types and hook props

* remove showStyleSelector prop, take it from hook now. Use styles as sent from backend, filtered and shaped there already

* add siteFragment to global interface for initial JP state, avoid linter warnings

* add changelogs

* remove the need for beta feature and gutenberg extension, if backend sends it, we use it

mreishus committed:
find_psr4_file: Quick exit if psr4_map is not populated (#39646)

matticbot committed:
Update storybook monorepo to v8.3.5 (#39669)

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Lock file maintenance (#39653)

* Lock file maintenance

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>

nateweller committed:
WAF: Include waf_supported and automatic_rules_last_updated in the jetpack/v4/waf REST API endpoint response (#39511)

fgiannar committed:
Jetpack Sync: Add 'woocommerce_analytics_first_activation' in options' whitelist (#39658)

* Jetpack Sync: Add 'woocommerce_analytics_first_activation' in options' whitelist

haqadn committed:
Image CDN: URL encode path parts (#39560)

* rawurlencode each path part in the image URL

* Fix a typo

* Add a test

* changelog

* Add comment to clarify space usage

---------

Co-authored-by: Peter Petrov <peter.petrov89@gmail.com>

darssen committed:
Sync: Don't sync set object terms action for blacklisted taxonomies (#39663)

* Filter out set_object_terms actions with blacklisted taxonomies

* Args in this flow does not have the terms retrieved as objects. Instead we rely directly on the taxonomy

* changelog

* Update projects/packages/sync/src/modules/class-terms.php

Co-authored-by: Foteini Giannaropoulou <giannaropoulou.foteini@gmail.com>

---------

Co-authored-by: Foteini Giannaropoulou <giannaropoulou.foteini@gmail.com>

simison committed:
Newsletter: change default reply-to option from "no reply" to "comments" (#39657)

ice9js committed:
Changelog and readme.txt edits. (#39668)

matticbot committed:
Update dependency express to v4.21.0 (#39670)

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>

jeherve committed:
Sharing: register menu even when user is not connected to wpcom. (#39677)

Fixes p1728349589696649-slack-CDD9LQRSN

anomiex committed:
explat: Fix Phan issues (#39443)

Add some missing `use` directives and add `WP_Error` to one return
declaration.

CodeyGuyDylan committed:
Fix/bug where connection banner shows in p2 (#39667)

* Fix issue where simple and atomic sites were showing connection banner

* changelog

CodeyGuyDylan committed:
Fix blogroll block unhelpful error when user not connected (#39638)

* Add connection nudge to blogroll block

* changelog

* Update connection banner to show if current user is not connected

elliottprogrammer committed:
My Jetpack: Fix recommendations VideoPress product card not showing "Purchase" button. (#39612)

CodeyGuyDylan committed:
Add/jetpack blocks add connection nudge where missing (#39647)

* changelog

* Add connection nudges to all payment blocks

* Check for current user connection instead of connected owner

CodeyGuyDylan committed:
Fix/payment blocks connection banner showing on simple sites (#39687)

* Fix connection banner for payment blocks on simple sites

* changelog

* Fix condition

kangzj committed:
Stats: Fix top post card on the Insight page (#39691)

* add required fields to endpoint

* changelog

* add discussion and post thumbnail

* comment

manzoorwanijk committed:
Social: Update the check to see if Bluesky is already connected (#39661)

darssen committed:
Add check to update post json api to handle null input type (#39676)

* Update class.wpcom-json-api-update-post-v1-2-endpoint.php to handle null input type

* changelog

xavier-lc committed:
Prevent welcome tour keyboard navigation from hijacking left right keys on the editor (#39683)

dilirity committed:
Sync my jetpack's Boost page with Boost's getting started page (#39130)

* Sync my jetpack's Boost page with Boost's getting started page

* add changelog

* Remove unused prop

* Update Boost pricing table to rely on the same data as my jetpack

* Add features list component to reduce code repetition

* Fix type incompatibility

* Remove unused code

* add changelog

* Reduce localized data

* Update premium pricing to rely on my jetpack variables instead of own

* Remove unused type

* Use data sync for 'my jetpack' data instead of directly localizing

* Add test for my jetpack data in Boost

* Fix test signature

* Update prop name

* Simplify tests

* Update php tests to be less strict in terms of feature order

* Add a check for feature count

CGastrell committed:
AI Client: change logo generator upgrade message (#39671)

* use getRedirectUrl package to get URL links to more info on usage and limits

* update upgrade copy edit for logo generator

* add changelogs

CGastrell committed:
AI Client: add image styles auto and none (#39689)

* add auto and none as available image styles

* changelog

CGastrell committed:
Jetpack AI: don't generate first logo without details (#39536)

* remove noisy debug call

* don't call first logo generation if there are no site details to work with

* make use logo button more prominent (secondary)

* add empty logo placeholder

* add empty logo history placeholder

* add changelog

* fix state checks for first load

* add enter key handler on prompt input

* improve empty placeholders

anomiex committed:
rolling release: Add timestamp to version numbers (#39701)

Apparently Atomic doesn't just pick the latest release, they also
require that the new version number compares greater than the old one as
measured by `version_compare()`. So we'll have to throw a timestamp in
after all.

nateweller committed:
WAF: Avoid using Waf_Rules_Manager from Waf_Runner::initialize() (#38944)

anomiex committed:
eslint: Enable more react rules (#39421)

* `react-hooks/exhaustive-deps`: Apply to `useSelect`.
* `react/no-direct-mutation-state`
* `react/no-find-dom-node`
* `react/no-unescaped-entities`

The first of those accounts for the majority of fixes in this PR, we had
a bunch of places either missing a dependency or specifying a dependency
unnecessarily. Of particular note is that constants defined outside of
the `render()` function, like `STORE_ID`, don't need to be dependencies.

Of the rest, `react/no-direct-mutation-state` was only violated in one
place, `react/no-unescaped-entities` in one storybook story, and
`react/no-find-dom-node` was already fixed everywhere.

anomiex committed:
jetpack: Render Slideshow block correctly in iframed editor (#39551)

When the block editor uses an iframe, our monolithic `editor.js` script
is only loaded and run at the top level, not inside the iframe. Thus,
any dynamic `import()` also only runs at the top level, and in
particular the loaded CSS does not affect the content inside the iframe.

The real solution to this would be to have the content-related code in
block.json "script" where it gets loaded everywhere (frontend, editor
top-level, and iframe) (unfortunately there's no option that runs only
on frontend+iframe, or only iframe) and run the code doing the
`import()` inside the iframe, so it adds the CSS inside the iframe. But
we can't really do that without a ton of refactoring that's largely
blocked on Gutenberg implementing lazy block loading.

So, in the mean time, let's configure mini-css-extract-plugin to add any
imported CSS into the iframe (if one exists) in addition to the top
level. That won't work for JS (e.g. the Map block, which has its own
hack), but that's ok for now.

matticbot committed:
Update wordpress monorepo (#39707)

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>

oskosk committed:
Goodreads Block: Make sure we check for id attribute in the goodreads block, when rendering (#39713)

* Make sure we check for id attribute in the goodreads block, before attempting to use it for rendering as id of the div.

* changelog

manzoorwanijk committed:
Social: Added Bluesky to social previews (#39659)

* Update dependencies

* Add Bluesky to social previews

* Add changelogs

* Update social-previews package to fix the Bluesky URL

* Update pnpm-lock.yaml

* Fix dependencies after rebase

fushar committed:
Admin bar: clean up WPCOM_ADMIN_BAR_UNIFICATION feature flag (#39692)

* Admin bar: clean up WPCOM_ADMIN_BAR_UNIFICATION

* changelog

darssen committed:
Added do_action_deprecated and apply_filters_deprecated to next version script (#39693)

darssen committed:
Infinite scroll added checks to validate input (#39618)

* Added isset checks to avoid Warnings

* changelog

* Added new isset for last_post_date

* Deprecate infinite_scroll_posts_where  hook

dilirity committed:
Boost: Fix page cache e2e tests (#39695)

* Fix not always logged in for page cache e2e test

This allows the page cache e2e test to be run independently every time
without issues.

* Use WP_Filesystem when trying to update wp-config.php for pace cache

* add changelog

* Use WP_Filesystem_Direct to update wp-config.php

* Set file permissions to avoid fatal errors

* Update param value

* Rename method

dilirity committed:
Critical CSS Gen: Add missing build folder for production (#39723)

* Add missing build folder to production build

* add changelog

dilirity committed:
Boost: Properly update minimum required WP version to 6.5 (#39719)

* Update minimum required version to 6.5

* add changelog

* Update changelog

jeherve committed:
Photon: avoid deprecation warnings when src is null (#39685)

* Photon: avoid deprecation warnings when src is null

In some scenarios, one can run into the following deprecation notices:

Deprecated: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in /wordpress/plugins/jetpack/13.9-a.7/jetpack_vendor/automattic/jetpack-image-cdn/src/class-image-cdn.php on line 682

This should avoid those notices.

* Check for the validity of the src earlier

See https://github.com/Automattic/jetpack/pull/39685#pullrequestreview-2357095359

---------

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

zinigor committed:
Jetpack Sync 3.14. (#39724)

phcp committed:
VideoPress: Add missing check for empty video_info on get_media_item_v1_1 (#39706)

* Add missing check for empty video_info

* changelog

* Use ! empty

dilirity committed:
Backport Critical css gen 1.0.1 (#39726)

lezama committed:
Newsletter: featured image settings UI defaults (#39700)

valterlorran committed:
Post-migration Experience: Add the plugin review task (#39711)

* Add the Review Plugins task

* changelog

* Add the calypso path for the task

---------

Co-authored-by: Valter Lorran <valter.lorran@automattic.com>

dilirity committed:
Backport boost 3.5.2-beta Changes (#39728)

darssen committed:
Added null check post status obj in sal site class (#39727)

* Add null check for post_status_obj to avoid PHP Warnings

* changelog

* Ensure we only assign get_post_status_object to the parent status in case the parent exists

anomiex committed:
eslint: Enable `@wordpress/no-base-control-with-label-without-id` rule (#39708)

When `<BaseControl>` is passed both a `label` and `id`, it makes an HTML
`<label>` so clicking the label focuses the contained element with the
corresponding `id`. Leaving off the `id` breaks that, so this rule flags
the situation for fixing.

It turns out, though, that everything it flagged in the monorepo was a
false positive.

A few cases we had something like
`<BaseControl label><TextControl /></BaseControl>`. There's no reason
for that when we can do `<TextControl label />` instead and get the
expected click-label-to-focus behavior.

The rest of the cases we were using `<BaseControl label>` to get a label
for something more complex, where there's really no `id` that makes
sense. In this case the documented thing to do is apparently
`<BaseControl><BaseControl.VisualLabel>label</BaseControl.VisualLabel></BaseControl>`
which does exactly the same thing that `<BaseControl label>` with no
`id` does internally.

anomiex committed:
eslint: Enable `react/jsx-key` rule (#39709)

This rule requires a `key` prop be specified on elements that it thinks
probably need one, namely elements inside arrays or returned from
iterators for `.map()` and the like.

In practice most of the cases fixed here are probably pointless, since
the arrays are never going to be sorted or modified at runtime. But it's
easier to add a key than to write a comment justifing ignoring it in
each case.

anomiex committed:
Use DEWP's new auto-wp-polyfill support (#39629)

`@wordpress/dependency-extraction-webpack-plugin` has added the ability
to detect a magic `/* wp:polyfill */` comment and only add `wp-polyfill`
as a dependency if that is found. Since `wp-polyfill` is just a custom
build of `core-js`, the intention is that
`babel-plugin-polyfill-corejs3` (maybe via `@babel/preset-env`) would be
used and then a custom Babel plugin would replace the `core-js` imports
with the magic comment.

This updates our Babel config to do just that, and removes the blanket
addition of `wp-polyfill` via `injectPolyfill` or otherwise.

matticbot committed:
Update dependency express to v4.21.1 (#39733)


---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>

Addison-Stavlo committed:
Jetpack-mu-wpcom plugin banner - update copies (#39725)

* update banner copies

* add changelog entry

* update changelog with correct page name

* enable fallback translations

* ensure jetpack-mu-wpcom domain for translation check

anomiex committed:
eslint: Remove `jsx-a11y/label-has-associated-control` override (#39736)

`@wordpress/eslint-plugin` sets `jsx-a11y/label-has-associated-control`
to "htmlFor", while we had it overridden to "either". Turns out there
was only place that wasn't already doing "htmlFor", so let's fix that
one place and go for it.

This also removes `jsx-a11y/label-has-for` which is deprecated in favor
of `jsx-a11y/label-has-associated-control`. The former config was
equivalent to "either".

nateweller committed:
WAF: Handle cases where outdated Waf_Rules_Manager has been autoloaded in standalone mode (#39652)

fgiannar committed:
Jetpack Sync: Move 'jetpack_plugins_updated' action to shutdown (#39717)

* Jetpack Sync: Move 'jetpack_plugins_updated' action to shutdown

arthur791004 committed:
Site Management Widget: Migrate to react (#39642)

* Site Management Widget: Migrate to react

* changelog

* Fix phan

* Fix lint

arthur791004 committed:
Revert "Site Management Widget: Migrate to react (#39642)" (#39741)

This reverts commit 00a566376634750f5448d350009330071b15c55d.

arthur791004 committed:
Site Management Widget: Migrate to react (2nd try) (#39742)

* Site Management Widget: Migrate to react

* changelog

* Fix phan

* Fix lint

* Restore class-wpcom-site-management-widget.php

darssen committed:
Update/check post exists under context in widget visibility module (#39746)

* Added check to ensure post exists before checking post_author

fgiannar committed:
Jetpack iCalendarReader: Fix Undefined array key Warnings (#39740)

* Jetpack iCalendarReader: Fix Undefined array key Warnings

CGastrell committed:
Jetpack AI: fix first logo notice (#39705)

* set new value for unlimited constant, accept both on type definitions

* abort first logo generation if user is over fair usage limit, show notice instead of error

* add changelogs

* fix store test

---------

Co-authored-by: Douglas <douglas.henri@automattic.com>

CGastrell committed:
AI Client: add style guessing and prompt processing (#39712)

* add type definitions as required by linter

* add guessStyle on use-logo-generator hook

* use guessStyle on logo generator prompt

* changelog

arthur791004 committed:
Site Management Widget: Remove unused files (#39743)

* Site Management Widget: Remove unused files

* changelog

mwatson committed:
AI Assistant: Make Write Brief no longer flag single quotes as spelling errors (#39735)

* AI Assistant: Make Breve no longer flag single quotes as spelling errors

* changelog

---------

Co-authored-by: Mike Watson <mike@MikeMBP.local>

sergeymitr committed:
Jetpack: optimize the 'admin_init' hook callback for Sharing module (#39737)

Jetpack sharing module: do not add the form processing callback to `admin_init` hook unless we actually need it.

CodeyGuyDylan committed:
Add/editor blocks add tracks for connection banner (#39732)

* Add tracks for editor connection banner

* changelog

* Fix connection banner Jetpack AI Search

kraftbj committed:
Related Posts: check for any single post without regard for post type. (#39730)

kraftbj committed:
Social Notes: add support for the ActivityPub reply-to block when the plugin is present (#39738)

nateweller committed:
Components: Migrate JetpackProtectLogo component from Jetpack Protect (#39703)

nateweller committed:
Components: Migrate MarkedLines component from Jetpack Protect (#39674)

jeherve committed:
Contact Form: improve security of the form endpoint (#39759)

* Contact Form: improve security of the form endpoint

Ensure that submitted forms can only be accessed by logged in users with the necessary capabilities.

Reference: p9dueE-8ng-p2

Co-authored-by:	Chris Jean <gaarai@gaarai.com>

* changelog

---------

Co-authored-by: Chris Jean <gaarai@gaarai.com>

fgiannar committed:
Jetpack Sync: Ensure Full Sync is only triggered on backend admin POST requests (#39747)

* Jetpack Sync: Ensure Full Sync is only triggered on backend admin POST requests

ice9js committed:
Backport jetpack 14.0-a.3 Changes (#39760)

* Changelog and readme.txt edits.

* Add changelog entry for Jetpack 13.9.1

zinigor committed:
Update Autoloader docs to include more limitations. (#39752)

* Added more docs about Autoloader limitations.

* Changelog.

* Added caveats and rephrased.

phcp committed:
VideoPress: Update VideoPress notice URL on Media Library (#39731)

* Update VideoPress notice URL on Media Library

* changelog

kraftbj committed:
Social Notes: ensure the filter does not impact other CPTs (#39763)

sixhours committed:
Post-migration Experience: Add "Connect your domain" task (#39680)

* Add domain connection task to post-migration checklist

* changelog

* Add task to list

* Add path on backend

* Add initialQuery to path if exists

* Linting

* Just testing something

* Check for primary domain mapping not WPCOM

* Undeclared function

* Fix spacing

* Linting fixes

* Fix site id

* Revert changes to is_completed callback

nateweller committed:
Components: Migrate DiffViewer from Protect (#39672)

kangzj committed:
JITM: Remove JITM from Jetpack Stats dashboard (Preparation to use embedded JITM) (#39714)

* expose functiont to render JITM and jitm env for stats

* changelog

* fix typo

* rever formatting

* pass jitm pass to stats admin

* ver and minify

* ver and minify

* ensure it has an option to return early

* revert unnecessary chagnes

* use cache buster as jitm version

* add comments

* revert changes

* revert changes

* Delete projects/packages/jitm/changelog/update-integration-jitm-in-stats

lezama committed:
Newsletter: Fix class not found error  (#39765)

arcangelini committed:
Help Center: prevent excessive loading (#39688)

jeherve committed:
Sync: avoid conflicts with other plugins using the filter (#39681)

* Sync: avoid conflicts with other plugins using the filter

Fixes #39654

* Provide actual plugin data to filter

See https://github.com/Automattic/jetpack/pull/39681/files#r1793690662

* Update projects/packages/sync/src/modules/class-callables.php

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

---------

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

coder-karen committed:
Map block: Add absolute positioning for container only when Mapkit is used (#39744)

fgiannar committed:
Jetpack Sync: Update default Post Type Blacklist (#39770)

* Jetpack Sync: Update default Post Type Blacklist

dilirity committed:
Boost: Backport 3.5.2 changes (#39771)

michaeldcain committed:
Verbum Comments: fix gravatar width in nested comments for block themes (#39757)

* Only set a max-width to images in the comment content

* changelog

* Handle non-block theme comments

* Fix fallback :not syntax

* Update comment

simison committed:
Newsletters: allow skipping subscription modals with URL query arg (#39644)

* Newsletters: allow skipping modals with URL query param

* changelog

* More elegant by cleaning up the URL param

* Use replaceState instead of pushState to avoid leaving history

sixhours committed:
Post-migration experience: Add domain DNS task (#39764)

* Add domain dns mapping task

* changelog

* Fix linter

* Try relative URL

* Remove calypso path callback

nateweller committed:
Components: Migrate ThreatSeverityBadge from Jetpack Protect (#39758)

nateweller committed:
Protect: Add a 30 second timeout argument to the threat history request (#39766)

nateweller committed:
Protect: Convert useProtectData to TypeScript (#39778)

tbradsha committed:
Tests: Use Ubuntu 22.04 runner for PHP 7.0 tests (#39779)

* Pass runner from matrix to workflow

* Run PHP 7.0 tests on Ubuntu 22.04

zaguiini committed:
Map block: Do not render the actual map if in block preview mode (#39768)

* Map block: Use block editor store preview flag to determine whether or not to display the placeholder

* changelog

* Cover the map container with the preview image

* Update CHANGELOG wording

* Remove preview attribute in favor of flag coming from the block editor store

* Default to mapHeight and fall back to 400px

* Add attribute removal changelog entry

jeherve committed:
Related Posts: only enable on all singular views in block themes (#39784)

Fixes #39783

This is a follow-up to #39730. Instead of enabling related posts in all singular views for every theme, keep relying on the `jetpack_relatedposts_filter_enabled_for_request` filter to enable theme in classic themes. Block themes will get them for every singular view by default, but classic themes will continue to only get them only for posts.

anomiex committed:
Jetpack: Stats: Fix csv trailing newline handling (#39787)

When #39665 switched from incorrectly using `str_getcsv( $csv, "\n" )`
to `explode()`, a minor difference between the two was unnoticed: the
former does not produce an empty element at the end of the result on a
trailing newline, while the latter does.

Add an `rtrim` on the data to ensure there isn't a trailing newline.

andres-blanco committed:
Post migration tasklist: Add/ssl provisioned task (#39776)

* add SSL provisioned task

* changelog

* check for slug ending to enable/disable the task

* add . to domain check

* add a better way to get the url

* remove is_visible_callback, since it defaults to true and add default check for task completion

---------

Co-authored-by: Andrés Blanco <email@gmail.com>

gmjuhasz committed:
Social: Fix OG tag conflicts with Jetpack Beta (#39792)

* Add Jetpack Beta's Jetpack slug to conflicting plugins

* changelog

lezama committed:
Add/newsletter preview menu item (#39782)

kraftbj committed:
Jetpack 14.0-a.5 (#39793)

nateweller committed:
Changelog and readme.txt edits. (#39781)

valterlorran committed:
Post-migration Experience: Make the "Migrating the site" task complete by default (#39769)

* Make the Migrating the site task complete as default

* changelog

---------

Co-authored-by: Valter Lorran <valter.lorran@automattic.com>

m1r0 committed:
Admin bar: Add reader icon tooltip (#39803)

sergeymitr committed:
My Jetpack: fix the 'Missing site connection' notice (#39809)

My Jetpack will ask to establish user connection even if site connection is missing.
This PR will give the "missing site connection" notice more priority so it would show up on non-connected sites.

nateweller committed:
WAF: Handle undefined entrypoint constant (#39812)

sergeymitr committed:
SSO: optimize 'admin_notices' action callback (#39811)

Make SSO admin class only hook into `admin_notices` when necessary.

lezama committed:
Newsletter: Fix "subscription_emails_use_excerpt" api. (#39815)

manzoorwanijk committed:
Social | Initial State: Migrate URLs on Social admin page (#39797)

* Pass the URLs from backend

* Install dependency

* Update URLs on Social admin page

* Add changelog

* Update pnpm-lock.yaml

* Clean up

nateweller committed:
Changelog and readme.txt edits. (#39814)

manzoorwanijk committed:
Social | Initial State: Migrate URLs in the editor (#39799)

* Install dependency

* Update pnpm-lock.yaml

* Update usePublicizeConfig hook

* Update connections management URL for broken connection

* Update connections management URL for Manage connections button in the editor

* Update connections management URL for unsupported connections in the editor

* Update the URL for user connection

* Clean up

* Add changelog

* Fix unit tests in Jetpack

tbradsha committed:
Tests: Fix block performance tests (#39790)

* Use CommonJS import of lodash module

* Wording tweak

* Try 2: swap lodash for lodash-es

jeherve committed:
Jetpack outdated WP notices: avoid fatals on outdated WP versions (#39745)

jeherve committed:
Story Block: be more defensive to avoid error in block previews (#39807)

Fixes #39694

dkmyta committed:
Protect: Add/apply shared AdminSectionHero component (#39702)

manzoorwanijk committed:
Fix site specific features passed to front-end for simple sites (#39817)

* Add `Current_Plan::get_simple_site_specific_features()` method

* Pass the features for simple sites to the script data

* Create siteHasFeature utility

* Add changelog

haqadn committed:
Boost: Remove unused setup prompt (#39831)

manzoorwanijk committed:
Social | Fix Bluesky display name when it's not set in user profile (#39840)

coder-karen committed:
Story block: Prevent 'key is not a prop' warning in ProgressBullet component (#39822)

coder-karen committed:
Slideshow block: Modifying Slideshow block key indexes to prevent same key notice (#39821)

CGastrell committed:
Fix Jetpack AI features control types (#39826)

* add changelog

* fix types for featuresControl and add default value for logo-generator feature

fredrikekelund committed:
Added `studio-sync` WPCOM plan feature (#39843)

* Added studio-sync plan feature

* changelog

m1r0 committed:
Admin bar: Add help icon tooltip  (#39845)

nateweller committed:
Protect: remove unused usePolling argument from credentials query hook (#39780)

simison committed:
Subscriptions: add a floating subscribe button (#37722)

Co-authored-by: Przemek Kuliga <pkuliga@gmail.com>

CGastrell committed:
AI Client: fix default initial state setting (#39846)

* don't map/set initial state when fetch fails

* changelog

* use currentValue instead of current limit to assess for free users

* dont hide placeholder on focus when disabled

fgiannar committed:
Revert "Jetpack Sync: Add 'woocommerce_analytics_first_activation' in options' whitelist" (#39837)

* Revert "Jetpack Sync: Add 'woocommerce_analytics_first_activation' in options…"

This reverts commit 155c6a3ecddd9d016c19c864384bec12b2d7a3de.

fgiannar committed:
WPCOM_JSON_API_Upload_Media_v1_1_Endpoint: Fix Fatals from uploads with invalid media type (#39858)

* WPCOM_JSON_API_Upload_Media_v1_1_Endpoint: Fix Fatals from uploads with invalid media type

manzoorwanijk committed:
Social | Fix reconnection for broken Bluesky connections (#39844)

gmjuhasz committed:
Social: Fix Bluesky profile URL (#39849)

* Add the Bluesky profile URL to the connection management

* changelog

manzoorwanijk committed:
Initial State: Replace `isEnhancedPublishingEnabled` with `siteHasFeature` call (#39835)

* Replace isEnhancedPublishingEnabled with siteHasFeature call

* Use constants instead of strings

haqadn committed:
Boost: Don't add a posts page if front page is set but not the posts page (#39862)

simison committed:
Floating subscribe button: add source for stats (#39860)

CGastrell committed:
AI Client: make fetch error retry optional (#39848)

* make reload handler optional on fetch error notice

* changelog

* provide empty retry handler

* move condition process to modal initializer

* cleanup new open handler

CGastrell committed:
add test instructions for next release (#39866)

sergeymitr committed:
Stats: optimize the Admin Bar hook (#39857)

Optimize the Stats Admin Bar hook by narrowing down the callback.

manzoorwanijk committed:
Social: Fix Bluesky custom domain handle not being accepted (#39872)

ice9js committed:
Backport changes for Jetpack 14.0-a.7 (#39855)

* Changelog and readme.txt edits.

* Update changelog entries for clarity

* Update readme.txt

haqadn committed:
Boost: Remove URL override from image guide (#39874)

Co-authored-by: Peter Petrov <peter.petrov89@gmail.com>

anomiex committed:
cli: Fix `jetpack changelog add` on a release branch (#39869)

When running `jetpack changelog add` while on a release branch, the
script should not throw an error about attempting to access `.length` on
`undefined`.

anomiex committed:
changlogger: When amending, do not preserve default link (#39868)

When doing `changelogger write --amend`, we normally preserve the
prologue, epilogue, and link unless these are overridden on the command
line.

An edge case is when you're amending version 1.2.3-beta to turn it into
1.2.3 release, and the link is the default link like
`https://example.com/diff/1.2.2...1.2.3-beta`. In this situation we
don't want to keep that link, we want to replace it with the new default
`https://example.com/diff/1.2.2...1.2.3`.

Also, when building changelogger, we have to install before trying to run it.

kangzj committed:
Stats: Use option value instead of transient for cache buster (#39887)

manzoorwanijk committed:
Initial State: Migrate `hasPaidFeatures` with feature check on front-end (#39838)

* Initial State: Migrate `hasPaidFeatures` with feature check on front-end

* Replace hasPaidPlan by hasSocialPaidFeatures

manzoorwanijk committed:
Initial State: Nuke hasPaidPlan flag (#39839)

gmjuhasz committed:
Social: Fix Bluesky profile name connection management (#39889)

* Default to external_name if display is not available

* changelog

coder-karen committed:
Hooks: Ensure translation loading is not triggered too early. (#39841)

coder-karen committed:
General: indicate compatibility with upcoming WP 6.7 (#39786)

sergeymitr committed:
Blocks: initialize connection assets in Story block (#39870)

Initialize connection assets in the Story block to fix the block assets loading issue.

bor0 committed:
Fix special chars display in SMP (#39756)

* Fix special chars display in SMP

The Site Management Panel may break when special chars are included in the blog's name (quotes, ampersands, etc.). This commit fixes that.

* Add ENT_QUOTES

* Changelog

* Use wp_specialchars_decode instead

matticbot committed:
phan: Update wpcom stubs (#39795)

Co-authored-by: Phabricator Bot <phabricator@automattic.com>

anomiex committed:
Update to node 22 (#39886)

Calypso has updated to node 22, which means it's time for us to do so as
well.

coder-karen committed:
Contact Info: Rearranging how the Contact Info block register process is required. (#39892)

anomiex committed:
Jetpack: Fix some minor eslint lints (#39753)

Should be no functionality changes, unless some of the `react-hooks/`
things were obscure bugs.

* `curly` × 2: Missing `{}` around `if` bodies.
* `eqeqeq` × 11: Use `===` and `!==` for comparisons.
* `no-case-declarations` × 8: If a `case` has `const` or `let` inside,
  wrap it in a block for scoping.
* `no-extra-boolean-cast` × 1: `if ( !! foo )` → `if ( foo )`.
* `no-redeclare` × 2: A few useless `/* global JSON */`.
* `no-undef` × 3: Remove some unused variables.
* `no-unused-expressions` × 7: All ignored by updating
  `projects/plugins/jetpack/modules/.eslintrc.js`.
* `no-useless-escape` × 4: Remove unnecessary `\`s in regexes.
* `react-hooks/exhaustive-deps` × 3: Added missing deps.
  * One case needed a bit of refactoring.
* `react-hooks/rules-of-hooks` × 5: Used `useSelect` and
  `useResizeObserver` unconditionally, and defined one that needed a
  conditional outside of `render()`.
* `valid-typeof` × 2: Broken `undefined !== typeof window` check,
  changed to just use `window?` in later conditionals.
* `@wordpress/no-global-active-element` × 2: Use `ownerDocument`.

Commit:
AI Assistant: Write Brief should not flag words like '2nd' and '100th' as spelling errors. (#39880)

* AI Assistant: Write Brief should not flag words like '2nd' and '100th' as spelling errors.

* changelog

* Typescript expects isNaN to be passed a number

matticbot committed:
Update dependency svgicons2svgfont to v14 (#39893)

* Update dependency svgicons2svgfont to v14
* Update svg-to-font for new svgicons2svgfont version

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

agrullon95 committed:
Verbum Comments: Fix iframe resize logic (#39791)

* Add changes to fix Verbum comments resize logic

* changelog

anomiex committed:
Add `<9.12.0` to `.engines.pnpm` (#39897)

We're waiting on updating to pnpm 9.12+ due to the regression of
pnpm/pnpm#3935. Update `.engines.pnpm` so Renovate stops trying to use
9.12 and then failing the "Lock files are up to date" check.

manzoorwanijk committed:
Initial State: Replace `isSocialImageGeneratorAvailable` with feature check (#39836)

* Initial State: Replace isSocialImageGeneratorAvailable with feature check

* Use constants instead of strings

gmjuhasz committed:
Changelog and readme.txt edits. (#39903)

matticbot committed:
Update Eslint packages (#39898)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

CGastrell committed:
Jetpack AI: decouple modal prompt input for reusability (#39864)

* decouple prompt input from prompt component, export it to make it available for other modals

* allow passing down CTA button label as prop

* implement AiModalPromptInput on GP image generation modal

* changelog

* fix wee css issue

anomiex committed:
required-review: Avoid requesting reviews from bots (#39895)

Attempting to request a review from a bot account will likely fail, as
bot accounts are not "collaborators".

Fortunately a bot account is probably going to look like
`@something[bot]`, so we can look for that to exclude them without
having to make extra API queries.

gogdzl committed:
Merge remote-tracking branch 'origin/trunk' into fix/crm/3477-fix-export-crash-when-company-custom-field

gogdzl committed:
Make suffix shorter

gogdzl committed:
Remove duplicate 'key´

gogdzl committed:
Leave old key as this is not a problem and removing it will make it a lot more difficult to fix the problem

gogdzl commented:
Basically we are always calling `add_name_clash_suffix_if_needed` in our DAL layer, but we are using it only in `fix_name_clash_if_needed` which is only called in the export method.

This is not ideal, but I can't see other quick alternative as all the methods involved in this problem are huge and hard to read.

coder-karen reviewed (approved):
This works for me now, the changes look good.

cleacos reviewed (commented):
Nice. If it fixes the issue.
